pyccel/codegen/printing/fcode.py

```diff
@@ -2213,6 +2279,36 @@ def _print_ConstructorCall(self, expr):
         code = '{0}({1})'.format(name, code_args)
         return self._get_statement(code)
 
+    def _print_NumpyUfuncBase(self, expr):
+        type_name = type(expr).__name__
+        func_name = numpy_ufunc_to_fortran[type_name]
+        code_args = ', '.join(self._print(i) for i in expr.args)
+        code = '{0}({1})'.format(func_name, code_args)
+        return self._get_statement(code)
+
+    def _print_MathFunctionBase(self, expr):
+        type_name = type(expr).__name__
+        func_name = math_function_to_fortran[type_name]
+        code_args = ', '.join(self._print(i) for i in expr.args)
+        code = '{0}({1})'.format(func_name, code_args)
+        return self._get_statement(code)
+
+    def _print_NumpySqrt(self, expr):
+        arg = expr.args[0]
+        code_args = self._print(arg)
+        if arg.dtype is NativeInteger() or arg.dtype is NativeBool():
+            code_args = 'Real({})'.format(code_args)
+        code = 'sqrt({})'.format(code_args)
+        return self._get_statement(code)
+
+    def _print_MathSqrt(self, expr):
+        arg = expr.args[0]
+        code_args = self._print(arg)
+        if arg.dtype is NativeInteger() or arg.dtype is NativeBool():
+            code_args = 'Real({})'.format(code_args)
+        code = 'sqrt({})'.format(code_args)
```

Comment:
See above. Why do we have the dictionary math_ufunc_to_fortran and then not use it? Is sqrt a special case?

Reply:
yes

Reply:
See above